pkg/common/telemetry/metrics.go

```diff
@@ -132,41 +148,65 @@ func (m *MetricsImpl) EmitKey(key []string, val float32) {
 }
 
 func (m *MetricsImpl) IncrCounter(key []string, val float32) {
-	for _, s := range m.metricsSinks {
-		s.IncrCounter(key, val)
+	if m.enableTrustDomainLabel {
+		m.IncrCounterWithLabels(key, val, nil)
+	} else {
+		for _, s := range m.metricsSinks {
+			s.IncrCounter(key, val)
+		}
 	}
 }
 
 // IncrCounterWithLabels delegates to embedded metrics, sanitizing labels
 func (m *MetricsImpl) IncrCounterWithLabels(key []string, val float32, labels []Label) {
+	if m.enableTrustDomainLabel {
+		labels = append(labels, Label{Name: TrustDomain, Value: m.c.TrustDomain})
+	}
+
 	sanitizedLabels := SanitizeLabels(labels)
 	for _, s := range m.metricsSinks {
 		s.IncrCounterWithLabels(key, val, sanitizedLabels)
 	}
 }
 
 func (m *MetricsImpl) AddSample(key []string, val float32) {
-	for _, s := range m.metricsSinks {
-		s.AddSample(key, val)
+	if m.enableTrustDomainLabel {
+		m.AddSampleWithLabels(key, val, nil)
```

Comment:
what happens if we always call `m.AddSampleWithLabels(key, val, nil)`? even may that reduce the uses of all repeted ifs?